### PR TITLE
[App] Add rm one level below project level

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR enables to delete files from the Lightning AI Cloud Filesystem.

```python
lightning rm my_file.txt
```

Fixes #\<issue_number>

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda

### CODE REVIEW

File: src/lightning/app/cli/commands/rm.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+    if root == "/" or len(splits) == 1:
+        return _error_and_exit("rm at the project level isn't supported")
+
+    client = LightningClient(retry=False)
+    projects = client.projects_service_list_memberships()
+
+    project = [project for project in projects.memberships if project.name == splits[0]]
+
+    # This happens if the user changes cluster and the project doesn't exit.
```
Comment (ethanwharris) [inline src/lightning/app/cli/commands/rm.py:55]:
```suggestion
    # This happens if the user changes cluster and the project doesn't exist.
```

File: src/lightning/app/cli/commands/rm.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        return _error_and_exit('rm "." and ".." may not be removed')
+
+    if ".." in rm_path:
+        return _error_and_exit('rm ".." or higher may not be removed')
+
+    root = os.path.join(root, rm_path)
+    splits = [split for split in root.split("/") if split != ""]
+
+    if root == "/" or len(splits) == 1:
+        return _error_and_exit("rm at the project level isn't supported")
```
Comment (ethanwharris) [inline src/lightning/app/cli/commands/rm.py:48]: PR title says `Add rm at the project level` but this says it's not supported

**Review by ethanwharris — APPROVED** [review]

File: src/lightning/app/cli/commands/rm.py (hunk 2 above)
Comment (tchaton) [reply src/lightning/app/cli/commands/rm.py:48]: Screenshot 2023-02-14 at 11 05 47

Renamed ;)

File: src/lightning/app/cli/commands/rm.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+@click.argument("rm_path", required=True)
+@click.option("-r", required=False, hidden=True)
+@click.option("--recursive", required=False, hidden=True)
+def rm(rm_path: str, r: bool = False, recursive: bool = False) -> None:
+    """Delete files on the Lightning Cloud filesystem."""
+
+    root = _pwd()
+
+    if rm_path in [".", ".."]:
```
Comment (justusschock) [inline src/lightning/app/cli/commands/rm.py:38]:
```suggestion
    if rm_path in (".", ".."):
```
nit: no mutables :)

**Review by justusschock — APPROVED** [review]

**Pull request merged**